src/sql/util.py

```diff
@@ -157,6 +157,99 @@ def show_deprecation_warning():
     )
 
 
+def check_duplicate_arguments(cmd_from, args) -> bool:
+    """
+    Raises UsageError when duplicate arguments are passed to magics.
+    Returns true if no duplicates in arguments or aliases.
+    """
+    if len(args) <= 1:
+        return True
+
+    allowed_duplicates = {
+        "sql": ["-w", "--with", "--append", "--interact"],
+        "sqlplot": ["-w", "--with"],
+        "sqlcmd": [],
+    }
+
+    disallowed_aliases = {
+        "sql": {
```

Comment:
I didn't understand why we need to define all the options here? Whenever there is a duplicate argument we can raise error.

Reply:
## The Issue
Initially I only used `duplicate_args` to keep track of all the duplicates. This left a loophole that a single hyphen alias of a double hyphen argument could both be used together. This was not duplication of the same argument exactly, but was a duplication in terms of the same thing being achieved by both the `-` and `--` options. Whichever one would be used the latest, was the one whose values would be passed to the parser. 

Consider the following examples and what would happen if we did not handle this aliased duplication:
`%sql duckdb:// --alias db-one -A db-two` would result in the alias being set to db-two.
`%sqlplot histogram --table penguins --column body_mass_g -c bill_depth_mm` would set the column to `bill_depth_mm` instead of `body_mass_g`.
`%sqlcmd columns --table penguins --schema s1 -s2` would result in `sqlcmd` trying to return the schema `s2` instead of the schema `s1`. 

Before I implemented the logic to handle these cases, these would all be legal and not raise errors, which is not the behavior that we would want.

## The Solution
I had no way of getting the alias for each of the commands without manually writing it down which is why I created the `disallowed_aliases` dictionary. The `either_or` list allows us to find out if there was any aliased duplication and produce an error message accordingly.

@neelasha23 Let me know if you have any questions!